MaterialLib/SolidModels/SolidsBase.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+#define MATERIALLIB_SOLIDMODELS_SOLIDBASE_H_
+
+#include <memory>
+
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace MeshLib
+{
+class Element;
+}
+
+namespace Solids
+{
+template <int DisplacementDim>
+struct SolidsBase
+{
+    struct InternalVariables
+    {
+        virtual void pushBackState() = 0;
+    };
+
+    using KelvinVector = ProcessLib::KelvinVectorType<DisplacementDim>;
+    using KelvinMatrix = ProcessLib::KelvinMatrixType<DisplacementDim>;
+
+    // TODO Avoid copies of data:
```

Comment:
Yes. That one cannot be virtual because of the template... But the fixed size one is virtual.
But when the DisplacementDim is removed, all of this might change.